dis_snek/models/snek/wait.py

```diff
@@ -1,30 +1,29 @@
-from asyncio import Future
+from asyncio import Event
 from typing import Callable, Optional
 
 __all__ = ["Wait"]
 
 
 class Wait:
-    def __init__(self, event: str, checks: Optional[Callable[..., bool]], future: Future):
+    def __init__(self, event: str, checks: Optional[Callable[..., bool]], future: Event):
         self.event = event
         self.checks = checks
         self.future = future
 
     def __call__(self, *args, **kwargs) -> bool:
-        if self.future.cancelled():
+        if self.future.is_set():
             return True
 
         if self.checks:
             try:
                 check_result = self.checks(*args, **kwargs)
-            except Exception as exc:
-                self.future.set_exception(exc)
+            except Exception:
```

Comment:
Shouldn't we just let the exception propagate if there is one?

I don't like the idea of completely discarding it.